### PR TITLE
feat: unify chat model selection menus

### PR DESCRIPTION
**Pull request by jan-kubica**
## Summary

- share one model picker between both composer entry points so their model lists cannot drift
- expose provider-default reasoning levels, inline localized explanations, and advanced-level grouping
- align automatic selection, model rows, and skill-menu controls

## Summary by CodeRabbit

- **New Features**
  - Added a reusable chat model menu with search, provider details, routing labels and loading/empty states.
  - Added reasoning-effort choices, grouped by level, with provider defaults and helpful guidance.
  - Improved submenu search and automatic focus when menus open.
  - Model selections now clearly display the chosen model and reasoning effort.

- **Localisation**
  - Added translated model-selection and reasoning-effort guidance across supported languages.

- **UI Improvements**
  - Updated the prompts section icon for clearer visual distinction.

### CODE REVIEW

File: apps/web/src/components/chat/composer-submenu-search.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      />
+    </InputGroup>
+  </div>
+);
+
+export const focusSearchOnOpen = (
+  ref: React.RefObject<HTMLInputElement | null>,
+) => {
+  setTimeout(() => ref.current?.focus(), 0);
+};
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/components/chat/composer-submenu-search.tsx]: Moved every submenu-search caller to the controlled, cancellable focus hook and added regression tests for focus scheduling and cancellation.\n\nCC on behalf of jan-kubica

File: apps/web/src/components/chat/chat-model-options-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+          <EffortHelp />
+          <EffortSubmenu
+            efforts={option.reasoningEfforts}
+            providerDefaultEffort={option.defaultReasoningEffort}
+            onSelect={onSelect}
+            selected={selected}
+            value={displayedEffort}
+          />
+        </>
+      )}
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/components/chat/chat-model-options-menu.tsx]: Added an explicit empty-effort guard, so models without reasoning options render as non-submenu rows instead of opening an empty menu. The options projection now uses a stable empty constant.\n\nCC on behalf of jan-kubica

File: packages/scripts/src/model-catalog-capabilities.test.ts [hunk 3]
```diff
@@ -30,6 +31,30 @@ const DECLARED_TEMPERATURE_POLICIES: Record<string, TemperaturePolicy> = {
...
+            {
+              id: "anthropic/claude-opus-5",
+              reasoning: { default_effort: "high" },
+            },
+            { id: "model-without-reasoning" },
+            { id: 42, reasoning: { default_effort: "medium" } },
+          ],
+        }),
+      ),
+    ).toEqual({ "anthropic/claude-opus-5": "high" });
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/scripts/src/model-catalog-capabilities.test.ts:53]: Added the numeric invalid-default boundary case and verified the focused parser tests pass.\n\nCC on behalf of jan-kubica

File: apps/web/src/components/chat/composer-submenu-search.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        onKeyDown={(event) => {
+          if (!isMenuNavigationKey(event.key)) {
+            event.stopPropagation();
+          }
+        }}
+        placeholder={placeholder}
+        ref={ref}
+        size="sm"
+        value={value}
+      />
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/components/chat/composer-submenu-search.tsx:47]: Added an accessible label to the shared submenu search input, derived from its localized placeholder.\n\nCC on behalf of jan-kubica

File: apps/web/src/components/chat/chat-model-options-menu.tsx [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+  return (
+    <MenuSub>
+      <MenuSubTrigger
+        aria-label={t("chat.modelSelector.effortLabel")}
+        className="text-muted-foreground min-w-28 px-2 data-highlighted:bg-transparent [&>svg:last-child]:!ms-auto"
+      >
+        {displayedEffort === null
+          ? t("chat.modelSelector.effortValues.providerDefault")
+          : t(EFFORT_LABEL_KEY[displayedEffort])}
+      </MenuSubTrigger>
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/components/chat/chat-model-options-menu.tsx:323]: Separated the visible effort label from the help control’s accessible label. The row keeps its effort name and describes it through a uniquely identified help description.\n\nCC on behalf of jan-kubica

File: apps/web/src/i18n/langs/cs.json [hunk 6]
```diff
@@ -512,7 +512,8 @@
     },
     "modelSelector": {
       "autoDescription": "Automaticky vybere model a úroveň uvažování.",
-      "autoLabel": "Automaticky",
+      "autoLabel": "Auto",
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/i18n/langs/cs.json]: Restored the Czech Automaticky translation and removed the stale untranslated-value baseline entry. The i18n consistency check passes.

CC on behalf of jan-kubica
